src/stratigraphy/benchmark/score.py

```diff
@@ -80,95 +55,38 @@ def get_scores(
 
             if parse_text(layer.material_description.text) == "":
                 logger.warning("Empty string found in predictions")
-        tp = hits
-        fp = len(file_prediction.layers) - tp
-        fn = number_of_truth_values[filename] - tp
-
-        if tp:
-            precision = tp / (tp + fp)
-            recall = tp / (tp + fn)
-        else:
-            precision = 0
-            recall = 0
-        document_level_metrics["document_name"].append(filename)
-        document_level_metrics["precision"].append(precision)
-        document_level_metrics["recall"].append(recall)
-        document_level_metrics["F1"].append(f1(precision, recall))
-        document_level_metrics["Number Elements"].append(number_of_truth_values[filename])
-        document_level_metrics["Number wrong elements"].append(fn + fp)
-        try:
-            document_level_metrics["Depth_interval_accuracy"].append(depth_interval_hits / depth_interval_occurences)
-            depth_interval_accuracies.append(depth_interval_hits / depth_interval_occurences)
-        except ZeroDivisionError:
-            document_level_metrics["Depth_interval_accuracy"].append(None)
-
-    if len(document_level_metrics["precision"]):
-        overall_precision = sum(document_level_metrics["precision"]) / len(document_level_metrics["precision"])
-        overall_recall = sum(document_level_metrics["recall"]) / len(document_level_metrics["recall"])
-        try:
-            overall_depth_interval_accuracy = sum(depth_interval_accuracies) / len(depth_interval_accuracies)
-        except ZeroDivisionError:
-            overall_depth_interval_accuracy = None
-    else:
-        overall_precision = 0
-        overall_recall = 0
-
-    if overall_depth_interval_accuracy is None:
-        overall_depth_interval_accuracy = 0
-
-    if return_document_level_metrics:
-        return {
-            "F1": f1(overall_precision, overall_recall),
-            "precision": overall_precision,
-            "recall": overall_recall,
-            "depth_interval_accuracy": overall_depth_interval_accuracy,
-        }, pd.DataFrame(document_level_metrics)
-    else:
-        return {
-            "F1": f1(overall_precision, overall_recall),
-            "precision": overall_precision,
-            "recall": overall_recall,
-            "depth_interval_accuracy": overall_depth_interval_accuracy,
-        }
+        layer_metrics.metrics[filename] = Metrics(
+            tp=hits, fp=len(file_prediction.layers) - hits, fn=number_of_truth_values[filename] - hits
+        )
+        if depth_interval_occurences > 0:
+            depth_interval_metrics.metrics[filename] = Metrics(
+                tp=depth_interval_hits, fp=depth_interval_occurences - depth_interval_hits, fn=0
+            )
+
+    return layer_metrics, depth_interval_metrics
```

Comment:
Would it make sense to compute these metrics step by step? In general, I do believe that it is nice when a function does one thing at a time. However, in this case, I assume we would have to iterate through all files twice which does not go hand in hand with performance.

Reply:
I agree, that would be cleaner, and I think the affect on performance will not be noticeable. I'll split this into two methods.